app/models/program.rb

```diff
@@ -8,10 +8,15 @@ class Program < ActiveRecord::Base
   has_many :tracks, dependent: :destroy
   has_many :difficulty_levels, dependent: :destroy
   has_many :events, dependent: :destroy do
-    def workshops
+    def require_registration
       where(require_registration: true, state: :confirmed)
     end
 
+    def with_registration_open
+      where(require_registration: true, state: :confirmed).
+      map { |e| e if e.max_attendees && ((e.max_attendees == 0) || (e.max_attendees > e.registrations.count)) }.compact
```

Comment:
Dunno :/ 
(I need to check the registrations count of the event (e.registrations.count))